src/Server/HTTPHandler.h

```diff
@@ -69,6 +71,22 @@ class HTTPHandler : public HTTPRequestHandler
 
     CurrentMetrics::Increment metric_increment{CurrentMetrics::HTTPConnection};
 
+    // The request_context and the request_credentials instances may outlive a single request/response loop.
+    // This happens only when the authentication mechanism requires more than a single request/response exchange (e.g., SPNEGO).
+    std::unique_ptr<Context> request_context;
+    std::unique_ptr<Credentials> request_credentials;
```

Comment:
I understand why you need to keep `GSSAcceptorContext` in `request_credentials`, but it's unclear to me why you need to store `request_context` here. Just to make `GSSAcceptorContext`? You only need `ExternalAuthenticators` for this.

Reply:
This is needed for cases when there are multiple HTTP request-response trips within the same TCP connection, when the first request with SPNEGO token doesn't carry the full information, and subsequent request(s) should add parts of the final token.

Reply:
> subsequent request(s) should add parts of the final token

Those parts are added to `GSSAcceptorContext`, not to `Context`, aren't they?

Reply:
Yes, but the context is created at that time, and is being preserved.

Reply:
I mean what was wrong with the old logic, where `Context` was copied in `HTTPHandler::handleRequest()` and then passed to `HTTPHandler::processQuery()`?